Fix start logic.

mojit0 opened the pull request:
Fix startup logic to support broader range of containers and broken container startup issues.

mojit0 committed:
Fix intermittently failing functional test.

This test's comparison on hours would consistently fail…if it was 12:xx on the clock, otherwise it would pass.

mojit0 committed:
Add missing portion of test fix.

Adjust the logic to not produce a 0:xx pm value.

mojit0 committed:
Fix start logic.

Adjusted MojitoServer to remove start() call and replace with listen(). This
provides a more flexible way to adapt to different hosting container needs.

Adjust archetype server.js files to provide more information/templates.

mojit0 committed:
Fix index.js archetypes.

The index.js archetypes were still reflecting the pre-clean-startup version.

mojit0 committed:
Fix host defaulting.

Defaulting the host wasn't working properly. Adjusted to support putting in an explicit host, otherwise don't use one at all.

caridy commented:
Let's just make sure we document this and provide an easy way to upgrade. Probably a good opportunity to introduce `mojito upgrade` command to facilitate this type of changes in the future.

mojit0 committed:
Adjust index/server coordination for container startup.

The index.js file is key for some containers so redirected it to call in server.js rather than mojito module.

mojit0 commented:
@caridy The whole point is to make it Manhattan friendly :) I've tested it with the latest changes and it seems stable now with Manhattan deployment uncommented in the server.js file. The release notes for the proposed 0.4.7 will have to point to updating both the index.js and server.js files in any applications which are using the older versions.

mojit0 committed:
Fix jslint errors.

mojit0 committed:
Readjust index.js reference.
